raiden/network/proxies/token_network.py

```diff
@@ -708,63 +678,128 @@ def set_total_deposit(
+                    channel_identifier=channel_identifier,
+                    detail_for=partner,
+                    partner=self.node_address,
+                    block_identifier=given_block_identifier,
+                )
+                current_balance = self.token.balance_of(
+                    address=self.node_address, block_identifier=given_block_identifier
+                )
+            except ValueError:
+                # If `given_block_identifier` has been pruned the checks cannot be
+                # performed.
                 pass
+            except BadFunctionCallOutput:
+                raise_on_call_returned_empty(given_block_identifier)
+            else:
+                if channel_onchain_detail.state != ChannelState.OPENED:
+                    msg = (
+                        f"The channel was not opened at the provided block "
+                        f"({given_block_identifier}). This call should never have "
+                        f"been attempted."
+                    )
+                    raise RaidenUnrecoverableError(msg)
 
-            log_details = {
-                "node": to_checksum_address(self.node_address),
-                "contract": to_checksum_address(self.address),
-                "channel_identifier": channel_identifier,
-                "partner": to_checksum_address(partner),
-                "new_total_deposit": total_deposit,
-                "previous_total_deposit": previous_total_deposit,
-            }
-            with log_transaction(log, "set_total_deposit", log_details):
-                self._set_total_deposit(
-                    channel_identifier,
-                    total_deposit,
-                    partner,
-                    previous_total_deposit,
-                    token,
-                    log_details,
+                amount_to_deposit = total_deposit - our_details.deposit
+
+                if total_deposit <= our_details.deposit:
+                    msg = (
+                        f"Current total deposit ({our_details.deposit}) is already larger "
+                        f"than the requested total deposit amount ({total_deposit})"
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                total_channel_deposit = total_deposit + partner_details.deposit
+                if total_channel_deposit >= UINT256_MAX:
+                    raise RaidenUnrecoverableError("Deposit overflow")
+
+                channel_participant_deposit_limit = self.channel_participant_deposit_limit
+                if total_deposit > channel_participant_deposit_limit:
+                    msg = (
+                        f"Deposit of {total_deposit} is larger than the "
+                        f"channel participant deposit limit"
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                token_network_deposit_limit = self.token_network_deposit_limit
```

Comment:
This too could be a problem because of the implicit RPC request.

Reply:
Changed to a function which takes block identifier